src/Compilers/CSharp/Test/Emit3/FirstClassSpanTests.cs

```diff
@@ -7759,6 +7861,178 @@ static class C
+        CompileAndVerify(comp, expectedOutput: expectedOutput).VerifyDiagnostics();
+
+        comp = CreateCompilationWithSpanAndMemoryExtensions(source);
+        CompileAndVerify(comp, expectedOutput: expectedOutput).VerifyDiagnostics();
+    }
+
+    [ConditionalFact(typeof(CoreClrOnly))]
+    public void OverloadResolution_ReadOnlySpanVsReadOnlySpan_Params()
+    {
+        var source1 = """
+            using System;
+            static class C
+            {
+                public static void M(params ReadOnlySpan<string> arg) => Console.Write(1);
+                public static void M(params ReadOnlySpan<object> arg) => Console.Write(2);
+            }
+            """;
+
+        var source2 = """
+            C.M(new string[0]);
+            """;
+        CreateCompilationWithSpanAndMemoryExtensions([source1, source2], parseOptions: TestOptions.Regular13).VerifyDiagnostics(
+            // (1,3): error CS0121: The call is ambiguous between the following methods or properties: 'C.M(params ReadOnlySpan<string>)' and 'C.M(params ReadOnlySpan<object>)'
+            // C.M(new string[0]);
+            Diagnostic(ErrorCode.ERR_AmbigCall, "M").WithArguments("C.M(params System.ReadOnlySpan<string>)", "C.M(params System.ReadOnlySpan<object>)").WithLocation(1, 3));
+
+        var expectedOutput = "1";
+
+        var comp = CreateCompilationWithSpanAndMemoryExtensions([source1, source2], parseOptions: TestOptions.RegularNext);
+        CompileAndVerify(comp, expectedOutput: expectedOutput).VerifyDiagnostics();
+
+        comp = CreateCompilationWithSpanAndMemoryExtensions([source1, source2]);
+        CompileAndVerify(comp, expectedOutput: expectedOutput).VerifyDiagnostics();
+
+        var source3 = """
+            using System;
+            C.M("a");
+            C.M(new object());
+            C.M(new object[0]);
+            C.M(default(Span<string>));
+            C.M(default(Span<object>));
+            C.M(default(ReadOnlySpan<string>));
+            C.M(default(ReadOnlySpan<object>));
+            """;
+
+        expectedOutput = "1221212";
+
+        comp = CreateCompilationWithSpanAndMemoryExtensions([source1, source3], parseOptions: TestOptions.Regular13);
+        CompileAndVerify(comp, expectedOutput: expectedOutput).VerifyDiagnostics();
+
+        comp = CreateCompilationWithSpanAndMemoryExtensions([source1, source3], parseOptions: TestOptions.RegularNext);
+        CompileAndVerify(comp, expectedOutput: expectedOutput).VerifyDiagnostics();
+
+        comp = CreateCompilationWithSpanAndMemoryExtensions([source1, source3]);
+        CompileAndVerify(comp, expectedOutput: expectedOutput).VerifyDiagnostics();
+    }
+
+    [Fact]
+    public void OverloadResolution_SpanVsUserDefined()
+    {
+        var source = """
+            using System;
+
+            C.M(new string[0]);
+            C.M(default(Span<string>));
+
+            class C
+            {
+                public static implicit operator C(string s) => new C();
```

Comment:
>public static implicit operator C(string s) => new C(); [](http://example.com/codeflow?start=16&length=55)

Does this overload have any effect? (Or should the argument be `string[]` instead?)